app/src/main/java/cm/aptoide/pt/ads/AdsRepository.java

```diff
@@ -188,4 +188,14 @@ public static boolean validAds(GetAdsResponse getAdsResponse) {
                 sharedPreferences, connectivityManager, resources, versionCodeProvider)
                 .observe()));
   }
+
+  public Observable<List<MinimalAd>> getAdsFromFirstInstall(int limitOfAds) {
+    return mapToMinimalAds(GetAdsRequest.ofFirstInstall(idsRepository.getUniqueIdentifier(),
+        googlePlayServicesAvailabilityChecker.isAvailable(context),
+        partnerIdProvider.getPartnerId(), limitOfAds, accountManager.isAccountMature(), httpClient,
+        converterFactory,
+        qManager.getFilters(ManagerPreferences.getHWSpecsFilter(sharedPreferences)),
+        sharedPreferences, connectivityManager, resources, versionCodeProvider)
+        .observe());
+  }
```

Comment:
Would extract this class with the added method to your own flavour. Maybe extend vanilla's class and only add the method you need? Would that work? If so, do it for the other ones below. (GetStoreWidgetsRequestFactory and RequestFactory)